gravitee-plugin-core/src/main/java/io/gravitee/plugin/core/internal/AbstractPluginEventListener.java

```diff
@@ -127,17 +126,39 @@ private void deployPlugin(Plugin plugin, Map<String, List<Plugin>> resolvedDepen
 
     private record PluginKey(String id, String type) {}
 
-    private static class PluginComparator implements Comparator<Plugin> {
+    protected static class PluginComparator implements Comparator<Plugin> {
+
+        private final String loadFirst;
+
+        public PluginComparator(String loadFirst) {
+            this.loadFirst = loadFirst;
+        }
 
         @Override
         public int compare(Plugin o1, Plugin o2) {
-            Integer pos1 = pluginPriority.indexOf(o1.type());
-            Integer pos2 = pluginPriority.indexOf(o2.type());
+            int pos1 = pluginPriority.indexOf(o1.type());
+            int pos2 = pluginPriority.indexOf(o2.type());
 
             if (pos1 >= 0) {
                 if (pos2 >= 0) {
+                    if (pos1 == pos2) {
+                        if (
+                            loadFirst != null &&
+                            o1.manifest().type().equalsIgnoreCase(SECRET_PROVIDER) &&
```

Comment:
AbstractPluginEventListener should be agnostic of this kind of details. 
Could you explain the solution according to the initial exception ?

Reply:
After discussion we keep things as is